src/test/java/com/williamfiset/algorithms/datastructures/balancedtree/TreapTreeTest.java

```diff
@@ -0,0 +1,164 @@
+package com.williamfiset.algorithms.datastructures.balancedtree;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.TreeSet;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TreapTreeTest {
+
+  static final int MAX_RAND_NUM = +100000;
+  static final int MIN_RAND_NUM = -100000;
+
+  static final int TEST_SZ = 500;
+
+  private TreapTree<Integer> tree;
+
+  @Before
+  public void setup() {
+    tree = new TreapTree<>();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testNullInsertion() {
+    tree.insert(null);
+  }
+
+  @Test
+  public void testTreeContainsNull() {
+    assertThat(tree.contains(null)).isFalse();
+  }
+
+  @Test
+  public void LeftLeftCase() {
```

Comment:
Question: does it make sense to have all these rotation test cases? My understanding is that the Treap is not deterministic in its structure. I'm trying to understand why these tests even pass, can you enlighten me?

Reply:
Just to make sure, when you say 'not deterministic', I'm assuming you are referring to the randomly generated priorities? 

In these rotation test cases, I initialize each node with a priority and node value so I can force the rotations to occur. 

[My goal was to mimic the rotations in this video](https://youtu.be/BQ9NBNnkJ5M?t=258)

Hopefully this answers your question?

Reply:
Ohh I see. You're inserting nodes with a preset priority with the insert(T val, int priority) instead of insert(T val). Yes that will work.